src/Google/Ads/GoogleAds/Lib/OAuth2TokenBuilder.php

```diff
@@ -156,23 +157,48 @@ public function withImpersonatedEmail(?string $impersonatedEmail): self
      *
      * @return FetchAuthTokenInterface the created OAuth2 object that can fetch auth tokens
      */
-    public function build()
+    public function build(): FetchAuthTokenInterface
     {
-        $this->defaultOptionals();
+        $this->defaultOptionals(); // Currently does nothing.
+
+        $scopes = ['https://www.googleapis.com/auth/adwords'];
+
+        // 1. Attempt to use Application Default Credentials if enabled.
+        if ($this->useApplicationDefaultCredentials === true) {
+            error_log("Attempting to use Application Default Credentials...");
+            try {
+                // ApplicationDefaultCredentials::getCredentials() finds credentials
+                // from environment variables, gcloud CLI, or metadata server.
+                $credentials = ApplicationDefaultCredentials::getCredentials($scopes);
+                error_log("Successfully obtained credentials via Application Default Credentials.");
+                return $credentials; // Return successfully found ADC credentials.
+            } catch (\Exception $e) {
```

Comment:
we should use `DomainException` here, as that's what this method has defined in `@throws`
```suggestion
            } catch (\DomainException $e) {
```